src/cli/cmd/generate-delegator.report.ts

```diff
@@ -0,0 +1,56 @@
+import fs from "fs";
+import { join } from "path";
+
+import { readPaymentReport, writeDelegatorReport } from "src/fs-client";
+import { globalCliOptions } from "src/cli/global";
+
+import {
+  REPORTS_SUCCESS_PAYMENTS_DIRECTORY,
+  CUSTOM_DELEGATOR_REPORT_DIRECTORY,
+} from "src/utils/constants";
+
+import {
+  BasePayment,
+  DelegatorPayment,
+  EPaymentType,
+} from "src/engine/interfaces";
+import { sortBy } from "lodash";
+
+export const generateDelegatorReport = async (commandOptions) => {
+  await fs.readdir(REPORTS_SUCCESS_PAYMENTS_DIRECTORY, async (err, files) => {
```

Comment:
It may be worth to visit failed payments and report with `WARNING` to baker that some past payments are still in failed records.